models/user_conversation_mapping.go

```diff
@@ -0,0 +1,97 @@
+package models
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"strings"
+)
+
+// UserConversationMapping represents a user's relationship to a conversation
+type UserConversationMapping struct {
+	UserID         int64  `json:"user_id,omitempty"`
+	ConversationID int64  `json:"conversation_id,omitempty"`
+	Role           Role   `json:"role,omitempty"`
+	Nickname       string `json:"nickname,omitempty"`
+	Pending        bool   `json:"pending,omitempty"`
+	LastOpened     string `json:"last_opened,omitempty"`
+}
+
+// Role represents a user's access control rights in a conversation
+type Role string
+
+const (
+	// Owner is a role that only the original creator of a conversation can have
+	// and represents the highest level of privilege
+	Owner Role = "owner"
+
+	// Admin is a role that multiple non-creator users in a conversation can
+	// have and represents elevated privilege over regular users
+	Admin Role = "admin"
+
+	// User is a role that multiple non-creator users in a converation can have
+	// and represents the lowest level of privilege
+	User Role = "user"
+
+	mappingsTable string = "users_to_conversations"
+)
+
+// CreateUserConversationMapping adds a row to the "users_to_conversations" table
+func (db *DB) CreateUserConversationMapping(mapping *UserConversationMapping) error {
+	var b strings.Builder
+	fmt.Fprintf(&b, "INSERT INTO %s(UserID, ConversationID, Role, Nickname, Pending) ", mappingsTable)
+	pendingFlag := 0
+	if mapping.Pending {
+		pendingFlag = 1
+	}
+	fmt.Fprintf(
+		&b,
+		"VALUES(%d, %d, %q, %q, %b)",
+		mapping.UserID,
+		mapping.ConversationID,
+		mapping.Role,
+		mapping.Nickname,
+		pendingFlag,
+	)
+
+	res, err := db.Exec(b.String())
+	if err != nil {
+		return err
+	}
+	if rowCount, err := res.RowsAffected(); err == nil {
```

Comment:
Does this error just not matter enough to be handled or logged?

Reply:
So the thing with this error is that even if `res.RowsAffected()` returns an error, the `INSERT` still succeeded, so I didn't want to make the function return an error, when the stuff that actually matters already succeeded. That being said, the error should at least be logged, yeah.